### PR TITLE
ref(api): Move additional API code to `snuba.api` and add `snuba.wsgi` entrypoint

### PR DESCRIPTION
**Pull request by tkaemming**
This change serves two purposes:

1. This moves API-specific* methods, static media, and templates to `snuba.api` and out of the project root,
2. This removes the implicit module-level initialization for logging and Sentry to a path where it will only be invoked by the uwsgi entrypoint. This prevents conflicts with the CLI where the effective log level was dependent on import ordering.

\* Well, kind of API-specific: you'll note places in the codebase where we are calling view code for non-API usage.

### CODE REVIEW

File: snuba/subscriptions/data.py [hunk 1]
```diff
@@ -19,31 +19,34 @@ class Subscription:
...
-                'conditions': self.conditions + [[["ifnull", ["offset", 0]], "<=", offset]],
-                'aggregations': self.aggregations,
-                'from_date': (timestamp - self.time_window).isoformat(),
-                'to_date': timestamp.isoformat(),
+                "project": self.project_id,
+                "conditions": self.conditions
+                + [[["ifnull", ["offset", 0]], "<=", offset]],
+                "aggregations": self.aggregations,
+                "from_date": (timestamp - self.time_window).isoformat(),
+                "to_date": timestamp.isoformat(),
```
Comment (tkaemming) [inline snuba/subscriptions/data.py]: #674

File: snuba/api/views.py [hunk 2]
```diff
@@ -287,7 +278,9 @@ def dataset_query(dataset, body, timer):
     assert http_request.method == "POST"
     ensure_table_exists(dataset)
 
-    schema = RequestSchema.build_with_extensions(dataset.get_extensions(), HTTPRequestSettings)
+    schema = RequestSchema.build_with_extensions(
+        dataset.get_extensions(), HTTPRequestSettings
+    )
```
Comment (tkaemming) [inline snuba/api/views.py]: #674

**Review by fpacifici — COMMENTED** [review]
DO you plan to move the non-api endpoints that are in views.py in a separate file out of /api?
Talking of /config and /dashboard.

Also did you double check the config and the dashboard still work. I ran into issues when I tried to move views to api last time. (well in the case of dashboard, please check it is still broken in the same way ....)

**Comment by tkaemming** [comment]
> DO you plan to move the non-api endpoints that are in views.py in a separate file out of /api?
> Talking of /config and /dashboard.

Good question. I'm not sure. I was thinking that this module was maybe inappropriately named. It might make more sense to call this `snuba.web` or `snuba.http` instead. Maybe a layout like this would make more sense:

```
/snuba
  /web
    /static
    /templates
    /views
      /api.py (query endpoints)
      /admin.py (config, dashboard, etc.)
    /wsgi.py
```

I'm not sure where `split.py` and `query.py` belong since they are getting less coupled to the HTTP responses.

> Also did you double check the config and the dashboard still work. I ran into issues when I tried to move views to api last time. (well in the case of dashboard, please check it is still broken in the same way ....)

**Comment by tkaemming** [comment]
Going to split this into several changes: 

1. move `snuba.api` to `snuba.web`, including the views, static assets, and templates
2. break up the views into modules instead of one 500 line file
3. move the wsgi entrypoint to improve the startup process

**Pull request closed without merging**